cmake/thirdparty/SetupOpenMP.cmake

```diff
@@ -10,7 +10,15 @@
 
 find_package(OpenMP REQUIRED)
 
-# avoid generator expressions if possible, as generator expressions can be 
+# check if the openmp flags used for C/C++ are different from the openmp flags
+# used by the Fortran compiler
+set(BLT_OPENMP_FLAGS_DIFFER FALSE CACHE BOOL "")
+if (ENABLE_FORTRAN)
+  string(COMPARE NOTEQUAL "${OpenMP_CXX_FLAGS}" "${OpenMP_Fortran_FLAGS}"
+         BLT_OPENMP_FLAGS_DIFFER )
+endif()
+
+# avoid generator expressions if possible, as generator expressions can be
 # passed as flags to downstream projects that might not be using the same
 # languages. See https://github.com/LLNL/blt/issues/205
 if (ENABLE_FORTRAN AND NOT OpenMP_CXX_FLAGS STREQUAL OpenMP_Fortran_FLAGS)
```

Comment:
```cmake
if (ENABLE_FORTRAN AND NOT OpenMP_CXX_FLAGS STREQUAL OpenMP_Fortran_FLAGS)
```

Can this whole line be replaced by the new variable? 
E.g. 
```cmake
if(BLT_OPEMP_FLAGS_DIFFER)
```

Reply:
Yes -- upon closer inspection, it looked like the older code was setting another variable, `ESCAPE_FORTRAN`. Since we are defining the `BLT_OPENMP_FLAGS_DIFFER` cache variable now, that variable is no longer needed. I changed those checks accordingly.